packages/components/src/autocomplete/autocompleter-ui.js

```diff
@@ -90,3 +98,21 @@ export function getAutoCompleterUI( autocompleter ) {
 
 	return AutocompleterUI;
 }
+
+function useOnClickOutside( ref, handler ) {
+	useEffect( () => {
+		const listener = ( event ) => {
+			// Do nothing if clicking ref's element or descendent elements
+			if ( ! ref.current || ref.current.contains( event.target ) ) {
+				return;
+			}
+			handler( event );
+		};
+		document.addEventListener( 'mousedown', listener );
+		document.addEventListener( 'touchstart', listener );
+		return () => {
+			document.removeEventListener( 'mousedown', listener );
+			document.removeEventListener( 'touchstart', listener );
+		};
+	}, [ ref, handler ] );
```

Comment:
refs should not be included in a hook's dependency list, as it could lead to unexpected behaviours. Let's add a comment nonetheless, since ESLint is not always able to detect that `ref` is a ref object

```suggestion
		// Disable reason: `ref` is a ref object and should not be included in a
		// hook's dependency list.
		// eslint-disable-next-line react-hooks/exhaustive-deps
	}, [ handler ] );
```

Reply:
Okay, sounds good to me.